src/core/eme/session.ts

```diff
@@ -268,38 +279,17 @@ function createSession(
       // If that was the case though, we should subscribe here.
       $loadedSessions.deleteAndClose(session);
       $storedSessions.delete(initData);
-    })
-    .publish();
+    });
 
-  return { session, sessionEvents };
+  return sessionEvents;
 }
 
-/**
- * Create a MediaKeySession and manage it (generate the request, manage
- * communications...).
- * @param {MediaKeys} mediaKeys
- * @param {Object} keySystem
- * @param {string} sessionType - Either "persistent-license" or "temporary"
- * @param {string} initDataType
- * @param {UInt8Array} initData
- * @param {Subject} errorStream
- * @returns {Observable}
- */
-function createSessionAndKeyRequest(
-  mediaKeys: IMockMediaKeys|MediaKeys,
-  keySystem: IKeySystemOption,
-  sessionType: MediaKeySessionType,
-  initDataType: string,
-  initData: Uint8Array,
-  errorStream: ErrorStream
-): Observable<Event|ISessionEvent> {
-  const {
-    session,
-    sessionEvents,
-  } = createSession(mediaKeys, sessionType, keySystem, initData, errorStream);
-
-  $loadedSessions.add(initData, session, sessionEvents);
-  log.debug("eme: generate request", initDataType, initData);
+export function generateKeyRequest(
```

Comment:
Add JSDOC please

Reply:
Also why exporting?

Reply:
And the return type

Reply:
Also Observable.defer could ensure that any side-effect is done at subscription only